File: trust-log.pl
#!/usr/bin/perl

use strict;
use warnings;

if (defined($ENV{'RELAYCLIENT'}) || defined($ENV{'TRUSTCLIENT'})) {
  print STDERR getppid()." mail from: $ENV{'SMTPMAILFROM'}\n";
  print STDERR getppid()." rcpt to: $ENV{'SMTPRCPTTO'}\n";
  print "\n";
}

File: trust-log.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

int main() {
  if ( (getenv("RELAYCLIENT")) ||
       (getenv("TRUSTCLIENT")) ) {
    fprintf(stderr, "%d mail from: %s\n", getppid(), getenv("SMTPMAILFROM"));
    fprintf(stderr, "%d rcpt to: %s\n", getppid(), getenv("SMTPRCPTTO"));
  }

  printf("\n");
  exit(0);
}
